File: lib/generateMd.js
import fs from 'fs/promises'
import config from '../config.js'

const { MD_OUPUT_PATH, DOCUMENT_TITLE } = config

export default async function generateMd (data) {
  const groups = data

  await fs.writeFile(MD_OUPUT_PATH, `# ${DOCUMENT_TITLE}\n\n`)

  for (const group of groups) {
    await fs.appendFile(MD_OUPUT_PATH, `## ${group.title}\n`)

    for (const album of group.albums) {
      let entry = '* '

      if (album.is_highlighted) entry += '**'
      entry += `${album.title} `

      for (const [index, artist] of album.artists.entries()) {
        if (index === 0) entry += '- '
        entry += artist.name
        if (index < album.artists.length - 1) entry += ','
      }

      if (album.year) entry += ` (${album.year})`
      if (album.is_highlighted) entry += '**'
      entry += ' '

      if (album.youtube) entry += `[youtube](${album.youtube}) `
      if (album.spotify) entry += `[spotify](${album.spotify}) `

      await fs.appendFile(MD_OUPUT_PATH, entry + '\n')
    }

    await fs.appendFile(MD_OUPUT_PATH, '\n')
  }

  let colophone = '&nbsp;\n&nbsp;\n&nbsp;\n'
  colophone += '### Curated by\n'
  colophone += 'Twitter: @fthrrhmn31  \n'
  colophone += 'Spotify: Aditya Fathurrahman \n'

  await fs.appendFile(MD_OUPUT_PATH, colophone)

  return MD_OUPUT_PATH
}

File: lib/generate_md.py
import sys
sys.path.append('../')
from config import config

def generate_md(data):
    """Writes the specified content to a file.

    Args:
        data

    Raises:
        PermissionError: If permission is denied to write to the file.
    """

    file_path = config["MD_OUTPUT_PATH"]
    groups = data
    try:
        with open(file_path, 'w') as f:
            f.write(f"# {config['DOCUMENT_TITLE']}\n\n")
            
            for group in groups:
                f.write(f"## {group['title']}\n")
                
                for album in group['albums']:
                    entry = "* "
                    
                    if (album['is_highlighted']): entry += "**"
                    
                    entry += album['title']
                    entry += " "
                    
                    artists = album['artists']
                    for index, artist in enumerate(artists):
                        if (index == 0): entry += "- "
                        entry += artist['name']
                        if (index < len(artists) - 1): entry += ", "
                    
                    if (album["year"]): entry += f" ({album['year']})"
                    
                    if (album['is_highlighted']): entry += "**"
                    entry += " "
                    
                    if (album["youtube"]): entry += f"[youtube]({album['youtube']}) "
                    if (album["spotify"]): entry += f"[spotify]({album['spotify']}) "
                    if (album["yes_no_wave"]): entry += f"[yes_no_wave]({album['yes_no_wave']}) "
                    
                    f.write(f"{entry}\n")
                
                f.write("\n")

            colophone = "&nbsp;\n&nbsp;\n&nbsp;\n"
            colophone += "### Curated by\n"
            colophone += "Twitter: @fthrrhmn31  \n"
            colophone += "Spotify: Aditya Fathurrahman \n"
            
            f.write(colophone)
            
        print(f"Content written successfully to: {file_path}")
    except PermissionError:
        print(f"Permission denied: {file_path}")

